features/filesystem/sd/SDBlockDevice.cpp

```diff
@@ -477,6 +509,57 @@ int SDBlockDevice::_cmd8() {
     return -1; // timeout
 }
 
+/* SDBlockDevice::_go_idle_state()
+ *
+ * ARGUMENTS
+ *  None
+ * DETAILS:
+ *  Put the SDCard into the SPI Mode idle state by sending the CMD0
+ *  (GO_IDLE_STATE) command. See the notes in the "SPI Startup" section
+ *  of the comments at the head of this file.
+ *
+ * RETURN:
+ *  -1              an error occured e.g. a valid response was not received.
+ *  R1_IDLE_STATE   (0x1), the successful response from CMD0.
+ */
+int SDBlockDevice::_go_idle_state() {
+    _spi.lock();
+    _cs = 0;
+    int cmd_arg = 0;    /* CMD0 argument is just "stuff bits"*/
+
+    /* Reseting the MCU SPI master may not reset the on-board SDCard, in which
+     * case when MCU power-on occurs the SDCard will resume operations as
+     * though there was no reset. In this scenario the first CMD0 will
+     * not be interpretted as a command and get lost. For some cards retrying
+     * the command overcomes this situation. */
+    for (int num_retries = 0; num_retries < SD_CMD0_GO_IDLE_STATE_RETRIES; num_retries++) {
+        /* send a CMD0 */
+        _spi.write(0x40 | SD_CMD0_GO_IDLE_STATE);
+        _spi.write(cmd_arg >> 24);
+        _spi.write(cmd_arg >> 16);
+        _spi.write(cmd_arg >> 8);
+        _spi.write(cmd_arg >> 0);
+        _spi.write(0x95);
+
+        // wait for the response (response[7] == 0)
+        for (int i = 0; i < SD_COMMAND_TIMEOUT; i++) {
+            int response = _spi.write(0xFF);
+            /* Explicitly check for the R1_IDLE_STATE response rather that most significant bit
+             * being 0 because invalid data can be returned. */
+            if ((response == R1_IDLE_STATE)) {
+                _cs = 1;
```

Comment:
this is the same as on line 557 (just response has different value). Might be moved down and just handle response (valid or timeout, default value -1, overwrite if found).